xdsl/dialects/bigint.py

```diff
@@ -15,9 +33,262 @@ class BigIntegerType(ParametrizedAttribute, TypeAttribute):
 
 bigint = BigIntegerType()
 
+
+class BinaryOperation(IRDLOperation, abc.ABC):
+    lhs = operand_def(bigint)
+    rhs = operand_def(bigint)
+    result = result_def(bigint)
+
+    assembly_format = "$lhs `,` $rhs attr-dict `:` type($result)"
+
+    def __init__(
+        self,
+        operand1: Operation | SSAValue,
+        operand2: Operation | SSAValue,
+        result_type: Attribute | None = None,
+    ):
+        if result_type is None:
+            result_type = SSAValue.get(operand1).type
+        super().__init__(operands=[operand1, operand2], result_types=[result_type])
+
+
+@irdl_op_definition
+class AddOp(BinaryOperation):
+    name = "bigint.add"
+
+    traits = traits_def(
+        Pure(),
+        Commutative(),
+        SameOperandsAndResultType(),
+    )
+
+
+@irdl_op_definition
+class SubOp(BinaryOperation):
+    name = "bigint.sub"
+
+    traits = traits_def(
+        Pure(),
+        Commutative(),
+        SameOperandsAndResultType(),
+    )
+
+
+@irdl_op_definition
+class MulOp(BinaryOperation):
+    name = "bigint.mul"
+
+    traits = traits_def(
+        Pure(),
+        Commutative(),
+        SameOperandsAndResultType(),
+    )
+
+
+@irdl_op_definition
+class FloorDivOp(BinaryOperation):
+    name = "bigint.floordiv"
+
+    traits = traits_def(
+        Pure(),
+        SameOperandsAndResultType(),
+    )
+
+
+@irdl_op_definition
+class ModOp(BinaryOperation):
```

Comment:
Can you mention that the sign of the result takes the sign of the divisor?
This is different than in other languages such as Javascript for instance.